src/main/CMV.java

```diff
@@ -119,8 +119,24 @@ private Boolean lic6_calculate() {
         return false;
     }
     
-    private Boolean lic7_calculate() {
-        return false;
+    private int lic7_calculate() {
+        final int K_PTS = Parameters.K_PTS;
+        final double LENGTH1 = Parameters.LENGTH1;
+        if(datapoints.length < 3){
+            return 0;
+        }
+        for(int i = 0; i < datapoints.length; i++){
+            if(i + K_PTS > datapoints.length - 1){
+                return 0;
+            }
+            int j = i + K_PTS;
```

Comment:
I believe it should be j=i+K_PTS+1 cause they should be separated by that many intervening points. If i=0, j will be at index K_PTS and there would only be K_PTS - 1 points in between